src/RequestFormContainer.js

```diff
@@ -0,0 +1,215 @@
+import {
+  useState,
+} from 'react';
+import {
+  useIntl,
+} from 'react-intl';
+import {
+  cloneDeep,
+  isEmpty,
+  isString,
+  unset,
+} from 'lodash';
+import moment from 'moment-timezone';
+import PropTypes from 'prop-types';
+
+import RequestForm from './RequestForm';
+import {
+  getRequestLevelValue,
+  hasNonRequestableStatus,
+} from './utils';
+import {
+  fulfilmentTypeMap,
+  REQUEST_LEVEL_TYPES,
+  RESOURCE_TYPES,
+} from './constants';
+
+const RequestFormContainer = ({
+  parentResources,
+  request,
+  onSubmit,
+  ...rest
+}) => {
+  const {
+    requester,
+    requesterId,
+    item,
+  } = request || {};
+  const intl = useIntl();
+  const [selectedItem, setSelectedItem] = useState(item);
+  const [selectedUser, setSelectedUser] = useState({ ...requester, id: requesterId });
+  const [selectedInstance, setSelectedInstance] = useState(request?.instance);
+  const [isPatronBlocksOverridden, setIsPatronBlocksOverridden] = useState(false);
+  const [instanceId, setInstanceId] = useState('');
+  const [isErrorModalOpen, setIsErrorModalOpen] = useState(false);
+  const [blocked, setBlocked] = useState(false);
+
+  const setItem = (optedItem) => {
+    setSelectedItem(optedItem);
+  };
+
+  const setUser = (user) => {
+    setSelectedUser(user);
+  };
+
+  const setInstance = (instance) => {
+    setSelectedInstance(instance);
+  };
+
+  const setIsBlocked = (value) => {
+    setBlocked(value);
+  };
+
+  const setStateIsPatronBlocksOverridden = (value) => {
+    setIsPatronBlocksOverridden(value);
+  };
+
+  const setStateInstanceId = (id) => {
+    setInstanceId(id);
+  };
+
+  const hideErrorModal = () => {
+    setIsErrorModalOpen(false);
+  };
+
+  const showErrorModal = () => {
+    setIsErrorModalOpen(true);
+  };
+
+  const getPatronManualBlocks = (resources) => {
+    return (resources?.patronBlocks?.records || [])
+      .filter(b => b.requests === true)
+      .filter(p => moment(moment(p.expirationDate).format()).isSameOrAfter(moment().format()));
+  };
+
+  const getAutomatedPatronBlocks = (resources) => {
+    const automatedPatronBlocks = resources?.automatedPatronBlocks?.records || [];
+    return automatedPatronBlocks.reduce((blocks, block) => {
+      if (block.blockRequests) {
+        blocks.push(block.message);
+      }
+
+      return blocks;
+    }, []);
+  };
+
+  const processBlocking = () => {
+    const [block = {}] = getPatronManualBlocks(parentResources);
+    const automatedPatronBlocks = getAutomatedPatronBlocks(parentResources);
+
+    if ((block?.userId === selectedUser.id || !isEmpty(automatedPatronBlocks)) && !isPatronBlocksOverridden) {
+      return setIsBlocked(true);
+    }
+
+    return undefined;
```

Comment:
The `precessBlocking` function doesn't imply returning anything, the function name doesn't have prefixes like `get` and `find`, so I return `undefined` instead of `null`. However, I can't just write a `return` statement, eslint throws the error `Expected to return a value at the end of arrow function  consistent-return`.

Reply:
I believe we don't need to `return setIsBlocked(true)`, we can just `setIsBlocked(true)`. Am I right?
And then we can also remove `return undefined`. We should not have any eslint errors.

Reply:
I missed that we need to make the `return` in the `handleSubmit` function when the `isBlocked` state is true. I'll change it a little bit as it was before.